File: backend/src/main/java/com/legaltrack/service/SupportService.java
package com.legaltrack.service;

import com.legaltrack.dto.common.PagedResponse;
import com.legaltrack.dto.support.*;
import com.legaltrack.enums.SupportStatus;
import org.springframework.data.domain.Pageable;

public interface SupportService {
    SupportTicketDto createTicket(CreateSupportTicketRequest request);
    SupportTicketDto getTicketById(Long ticketId);
    PagedResponse<SupportTicketDto> getMyTickets(Pageable pageable);
    PagedResponse<SupportTicketDto> getAllTickets(SupportStatus status, Pageable pageable);
    SupportMessageDto addMessageToTicket(Long ticketId, SendSupportMessageRequest request);
    SupportTicketDto updateTicketStatus(Long ticketId, UpdateSupportStatusRequest request);
}

File: frontend/src/services/supportService.js
import api from './api';

export const supportService = {
  createTicket: async (data) => {
    const res = await api.post('/support/tickets', data);
    return res.data;
  },

  getTicketById: async (id) => {
    const res = await api.get(`/support/tickets/${id}`);
    return res.data;
  },

  getMyTickets: async (page = 0, size = 10) => {
    const res = await api.get(`/support/tickets/my?page=${page}&size=${size}`);
    return res.data;
  },

  getAllTickets: async (status, page = 0, size = 10) => {
    const res = await api.get(`/support/tickets?${status ? `status=${status}&` : ''}page=${page}&size=${size}`);
    return res.data;
  },

  addMessageToTicket: async (ticketId, messageData) => {
    const res = await api.post(`/support/tickets/${ticketId}/messages`, messageData);
    return res.data;
  },

  updateTicketStatus: async (ticketId, statusData) => {
    const res = await api.patch(`/support/tickets/${ticketId}/status`, statusData);
    return res.data;
  }
};
